### PR TITLE
Df asm reactions

### PR DESCRIPTION
**Pull request by dflanderka**
- implementation of new assembly algorithm in reactions

### CODE REVIEW

File: src/reaction/sorption_base.hh [hunk 1]
```diff
@@ -103,13 +106,34 @@ public:
...
 
     /// Input data set - fields in this set are read from the input file.
-    FieldSet input_data_set_;
+    FieldSet input_field_set_;
 
     /// Fields indended for output, i.e. all input fields plus those representing solution.
     EquationOutput output_fields;
 
+    /// Instances of FieldModel used in assembly methods
+    Field<3, FieldValue<3>::Scalar > scale_aqua;
```
Comment (jbrezmorf) [inline src/reaction/sorption_base.hh:115]: Slight optimization by using no_sorbing_surface_cond in calculation of scale sorbed.

File: src/reaction/sorption.cc [hunk 2]
```diff
@@ -118,23 +165,41 @@ SorptionMob::SorptionMob(Mesh &init_mesh, Input::Record in_rec)
...
+            fn_mob_surface_cond(), eq_fields_->porosity, eq_fields_dual_->immob_porosity_), 0.0);
 }
 
 
 /***********************************                   *****************************************************/
 /*********************************** SORPTION_IMMOBILE *****************************************************/
 /***********************************                   *****************************************************/
 
+// Functors for computing models
+struct fn_immob_scale_aqua {
```
Comment (jbrezmorf) [inline src/reaction/sorption.cc:183]: JB: Possible error in master, should be por_imm here.

File: src/reaction/sorption.cc [hunk 3]
```diff
@@ -58,32 +80,36 @@ const int SorptionSimple::registrar =
 SorptionSimple::~SorptionSimple(void)
 {}
 
-void SorptionSimple::compute_common_ele_data(const ElementAccessor<3> &elem)
+void SorptionSimple::init_field_models()
```
[the remark this answers is not in the document]
Comment (dflanderka) [reply src/reaction/sorption.cc:83]: It is not possible, method Field<>::set must be called after set_mesh.

File: src/reaction/sorption_base.hh (hunk 1 above)
Comment (dflanderka) [reply src/reaction/sorption_base.hh:115]: Done.

**Pull request merged**